inference-engine/src/vpu/graph_transformer/src/middleend/passes/annotate_memory_types.cpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "vpu/middleend/pass_manager.hpp"
+
+namespace vpu {
+
+namespace {
+
+template<class DataObjects>
+std::vector<vpu::MemoryType> gather(const DataObjects& dataObjects) {
+    std::vector<vpu::MemoryType> types;
+    types.reserve(dataObjects.size());
+    std::transform(dataObjects.begin(), dataObjects.end(), std::back_inserter(types), [](const Data& data) { return data->memReqs(); });
```

Comment:
Why `back_inserter`?

Reply:
I don't see other option assuming `types` is empty. Do you have something specific on your mind?

Reply:
It is only about design. It is ok for me